bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -89,17 +101,26 @@ pub struct MessageToXcm<
 	InboundQueuePalletInstance,
 	AccountId,
 	Balance,
+	ConvertAssetId,
+	UniversalLocation,
+	GlobalAssetHubLocation,
 > where
 	CreateAssetCall: Get<CallIndex>,
 	CreateAssetDeposit: Get<u128>,
 	Balance: BalanceT,
+	ConvertAssetId: MaybeEquivalence<TokenId, Location>,
+	UniversalLocation: Get<InteriorLocation>,
```

Comment:
What you are using here is actually `EthereumUniversalLocation`.

The established pattern across FRAME/Polkadot/Cumulus for `UniversalLocation` is to describe the universal (not relative) location of the current context. Used to define the context under which all other relative locations are evaluated from.

I see two options for cleaning current code up:
1. rename this `UniversalLocation` -> `EthereumUniversalLocation` (still anti-pattern if you ask me, but at least it's clear),
2. use `UniversalLocation`, populate it in runtime with the BH location, and have `pallet-inbound-queue` (deployed at BH) store locations relative to BH instead of relative to Ethereum (they can easily be reanchored to Ethereum on demand using the now correct `UniversalLocation`).

These now hold locations relative to BH, not Ethereum
```diff
-	/// Lookup table for foreign token ID to native location relative to ethereum
+	/// Lookup table for foreign token ID to native location
	#[pallet::storage]
	pub type ForeignToNativeId<T: Config> =
		StorageMap<_, Blake2_128Concat, TokenId, xcm::v4::Location, OptionQuery>;

-	/// Lookup table for native location relative to ethereum to foreign token ID
+	/// Lookup table for native location to foreign token ID
	#[pallet::storage]
	pub type NativeToForeignId<T: Config> =
		StorageMap<_, Blake2_128Concat, xcm::v4::Location, TokenId, OptionQuery>;
```

I honestly prefer option 2 since it follows the established pattern for all pallets and general location-handling code.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/5546#discussion_r1750553286